compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/codegen/MethodSignatureMapper.kt

```diff
@@ -74,7 +74,7 @@ class MethodSignatureMapper(private val context: JvmBackendContext) {
         if (nameForSpecialFunction != null) return nameForSpecialFunction
 
         val property = (function as? IrSimpleFunction)?.correspondingPropertySymbol?.owner
-        if (property != null) {
+        if (property != null && function.name.isSpecial) {
```

Comment:
Could you clarify why this is needed? It would seem that property accessor's name is always `isSpecial`

Reply:
For properties of inline class type (or in an inline class) the name is mangled in the inline class lowering and no longer special at this point. This wasn't an issue before, since we treated such accessors as ordinary functions.

I'm not actually sure why we have special names for properties in the first place. Would it make sense to put in the JvmAbi names after psi2ir?

Reply:
Thanks. That would make sense, yes. But since we don't lower lazy stub IR for external dependencies right now, this would help only for properties from the same module.

Reply:
In that case could we do it as part of a generator extension in psi2ir?

Reply:
Yes, an extension which is then passed to `ExternalDependenciesGenerator` much like `context.extensions.externalDeclarationOrigin` or `facadeClassGenerator` seems like a good idea!